### ISSUE TITLE
Custom Media Queries - Media queries level 5

### ISSUE THREAD

**Issue by armano2** [biomejs/biome#10855]
### Description

Currently parses will fail when provided with valid css `@custom-media`, this feature is farely new, but due to down-leveling by tools like postcss/lightningcss i'd like to be able to start using it with biome :)

https://www.w3.org/TR/mediaqueries-5/#custom-mq
https://developer.mozilla.org/en-US/docs/Web/CSS/Reference/At-rules/@custom-media

```css
@custom-media --medium-screen (min-width: 40em) and (max-width: 60em);

@media (--medium-screen) {
}
```

tests: https://wpt.fyi/results/css/mediaqueries/at-custom-media-parsing.html?label=experimental&label=master&aligned
